redhat-appstudio/service-provider-integration-operator
# experimental AWS Token Storage

### What does this PR do?
AWS token storage first implementation.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-342

### How to test this PR?


```
quay.io/redhat-appstudio/pull-request-builds:spi-operator-456
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-456
```

1. Deploy SPI with images ^^. You may delete Vault deployment.
2. If you've used AWS CLI, you should have `~/.aws/config` and `~/.aws/credentials` files on your local. You need to get these files into SPI pods, so create a secret in spi-system namespace `kubectl create secret generic aws-secretmanager-credentials --from-file=/Users/mvala/.aws/config --from-file=/Users/mvala/.aws/credentials -n spi-system`
3. update SPI operator and oauth deployments to mount the secret, you can also remove vault credentials mounts (like https://gist.github.com/sparkoo/9a8698519d9a52bf30391bde48353a64#file-deployment-patch)
4. edit configmap`spi-shared-environment-config` and add `TOKENSTORAGE=aws`, you can remove any Vault options
5. That should be all, restart the pods, try upload/oauth some tokens, delete them, update them... You should see the secrets with AWS cli
6. If AWS credentials are not set or are invalid, SPI must fail to start with proper message
